src/main/java/org/jabref/gui/groups/GroupDialogView.java

```diff
@@ -0,0 +1,184 @@
+
+import org.jabref.Globals;
+import org.jabref.JabRefGUI;
+import org.jabref.gui.BasePanel;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.util.BaseDialog;
+import org.jabref.gui.util.IconValidationDecorator;
+import org.jabref.gui.util.ViewModelListCellFactory;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.groups.AbstractGroup;
+import org.jabref.model.groups.GroupHierarchyType;
+import org.jabref.preferences.JabRefPreferences;
+
+import com.airhacks.afterburner.views.ViewLoader;
+import de.saxsys.mvvmfx.utils.validation.visualization.ControlsFxVisualizer;
+
+public class GroupDialogView extends BaseDialog<AbstractGroup> {
+
+    // Basic Settings
+    @FXML private TextField nameField;
+    @FXML private TextField descriptionField;
+    @FXML private TextField iconField;
+    @FXML private ColorPicker colorField;
+    @FXML private ComboBox<GroupHierarchyType> hierarchicalContextCombo;
+
+    // Type
+    @FXML private RadioButton explicitRadioButton;
+    @FXML private RadioButton keywordsRadioButton;
+    @FXML private RadioButton searchRadioButton;
+    @FXML private RadioButton autoRadioButton;
+    @FXML private RadioButton texRadioButton;
+
+    // Option Groups
+    @FXML private TextField keywordGroupSearchTerm;
+    @FXML private TextField keywordGroupSearchField;
+    @FXML private CheckBox keywordGroupCaseSensitive;
+    @FXML private CheckBox keywordGroupRegex;
+
+    @FXML private TextField searchGroupSearchTerm;
+    @FXML private CheckBox searchGroupCaseSensitive;
+    @FXML private CheckBox searchGroupRegex;
+
+    @FXML private RadioButton autoGroupKeywordsOption;
+    @FXML private TextField autoGroupKeywordsField;
+    @FXML private TextField autoGroupKeywordsDeliminator;
+    @FXML private TextField autoGroupKeywordsHierarchicalDeliminator;
+    @FXML private RadioButton autoGroupPersonsOption;
+    @FXML private TextField autoGroupPersonsField;
+
+    @FXML private TextField texGroupFilePath;
+
+    private final ControlsFxVisualizer validationVisualizer = new ControlsFxVisualizer();
+    private final GroupDialogViewModel viewModel;
+
+    public GroupDialogView(DialogService dialogService, BasePanel basePanel, JabRefPreferences prefs, AbstractGroup editedGroup) {
+        viewModel = new GroupDialogViewModel(dialogService, basePanel, prefs, editedGroup);
+
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        if (editedGroup == null) {
+            this.setTitle(Localization.lang("Add subgroup"));
+        } else {
+            this.setTitle(Localization.lang("Edit group") + " " + editedGroup.getName());
+        }
+
+        setResultConverter(viewModel::resultConverter);
+
+        getDialogPane().getButtonTypes().setAll(ButtonType.OK, ButtonType.CANCEL);
+
+        getDialogPane().getScene().getWindow().sizeToScene();
+    }
+
+    public GroupDialogView(DialogService dialogService, AbstractGroup editedGroup) {
```

Comment:
Can you please inline these constructors that use global variables (let the caller care where to get them).